pkg/kodo/cmd/deploy.go

```diff
@@ -67,6 +72,81 @@ func Deploy(deployVar *DeploymentVariables, envVar *EnvironmentVariables) error
 	} else {
 		log.Fatal(deploymentError)
 	}
+
 	return deploymentError
 
 }
+
+func Service(deployVar *DeploymentVariables, envVar *EnvironmentVariables) (*corev1.Service, error) {
+
+	client, clientError := newOpenShiftClient(envVar)
+
+	if clientError != nil {
+		log.Fatal(clientError)
+		return nil, clientError
+	}
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "app-image-service",
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port:       80, // use correct datatype, hint: int32
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr1.FromInt(int(deployVar.Port)), // port is to be obtained from the command flag.
+				},
+			},
+			Selector: map[string]string{
+				"app": "app-image-deployment",
+			},
+		},
+	}
+	_, serviceError := client.CoreV1().Services(envVar.Namespace).Create(context.TODO(), svc, metav1.CreateOptions{})
+	if serviceError == nil {
+		fmt.Printf("\nService created")
+	} else {
+		log.Fatal(serviceError)
+	}
+	return svc, serviceError
+
+}
+
+func Route(deployVar *DeploymentVariables, envVar *EnvironmentVariables, svc *corev1.Service) error {
+
+	_, clientError := newOpenShiftClient(envVar)
+
+	if clientError != nil {
+		log.Fatal(clientError)
+		return clientError
+	}
+
+	routeObj := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "myroute",
```

Comment:
To avoid naming conflicts, could we generate a unique name using the name of the image we are deploying?

We could do this across the project in a different PR. Could you track a separate issue for this?

Reply:
https://github.com/cli-playground/kodo/issues/18 Assigned to @jaideepr97